src/domain/safe/safe.repository.ts

```diff
@@ -119,7 +119,7 @@ export class SafeRepository implements ISafeRepository {
     const page: Page<MultisigTransaction> =
       await transactionService.getMultisigTransactions(
         safeAddress,
-        '-modified',
+        'nonce',
```

Comment:
Why was this changed?

Reply:
Just copying the Rust implementation, which is [ordering by nonce](https://github.com/safe-global/safe-client-gateway/blob/2534c54ca58878fcf0da9188896b0643505fddc7/src/routes/transactions/handlers/queued.rs#L36). I assumed this was a bug, wasn't it?

Reply:
See https://github.com/safe-global/safe-client-gateway/blob/2534c54ca58878fcf0da9188896b0643505fddc7/src/routes/safes/handlers/safes.rs#L122-L130

Reply:
Not sure I'm following 🙂 [That function](https://github.com/safe-global/safe-client-gateway/blob/2534c54ca58878fcf0da9188896b0643505fddc7/src/routes/safes/handlers/safes.rs#L122-L130) gets the last queued transaction, which if I'm not wrong will be used to populate the payload of the Safe Info endpoint. The function I modified gets a list of queued transactions. 

So if I'm not wrong the function I modified needs to call `transactionService.getMultisigTransactions` with `nonce` as `ordering`, and for function you pointed we'd need to implement another function which need to call `transactionService.getMultisigTransactions` with `-modified` as `ordering`. Am I missing something?

Reply:
Another option would be parametrizing the call and just passing the `ordering` as parameter. Not sure which is the clearer solution, what do you think?

Reply:
Yes, I agree that's the best option. So do you think using this `getQueuedTransactions` function to retrieve the transactions ordered by `nonce` is ok, or should we split it into two functions in this PR? 🙂

Reply:
We should split into two functions in this PR. `getTransactionQueue` with the modified parameter is already being used by a local change of mine.

Reply:
Perfect, but for what purpose are you using this function in your local changes? I mean, in the Rust implementation I see two purposes: 

- Get the last transaction of the queue.
- Get the list of queued transactions.

Which names do you propose?  Do you plan to maintain `getTransactionQueue` for just picking the last transaction in the queue? 🙂

Reply:
My suggestion would be the following:

```typescript
async getTransactionQueueByModified( // new function for `ISafeRepository`
  chainId: string,
  safeAddress: string,
  limit?: number,
  offset?: number,
): Promise<Page<MultisigTransaction>> {
  return this.getTransactionQueue(
    chainId,
    safeAddress,
    '-modified',
    limit,
    offset,
  );
}

async getTransactionQueueByNonce( // new function for `ISafeRepository`
  chainId: string,
  safeAddress: string,
  limit?: number,
  offset?: number,
): Promise<Page<MultisigTransaction>> {
  return this.getTransactionQueue(
    chainId,
    safeAddress,
    'nonce',
    limit,
    offset,
  );
}

private async getTransactionQueue( // this becomes private. Remove from `ISafeRepository`
  chainId: string,
  safeAddress: string,
  ordering: string, // new ordering parameter
  limit?: number,
  offset?: number,
): Promise<Page<MultisigTransaction>> {
  ...
}
```

Wdyt?

Reply:
Yes, that's a great solution 👍🏻 I was thinking of having something like `getLastQueuedTransaction`, but maybe `getTransactionQueueByModified` makes it more reusable.

I will make the change, thanks.

Reply:
> getLastQueuedTransaction

In my opinion, `getLastQueuedTransaction` is so specific that I left it in the route level (local changes) 😉

Reply:
Changed in [1ca115c](https://github.com/5afe/safe-client-gateway-nest/pull/215/commits/1ca115cafd135af939cb823dff24f26b7e56151a)